cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -240,15 +246,17 @@ protected Future<Void> apiSecret() {
         if (cruiseControl != null) {
             return secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()))
                     .compose(oldSecret -> {
-                        Secret newSecret = cruiseControl.generateApiSecret();
+                        Secret newSecret = cruiseControl.generateApiSecret(passwordGenerator);
 
                         if (oldSecret != null)  {
                             // The credentials should not change with every release
                             // So if the secret with credentials already exists, we re-use the values
                             // But we use the new secret to update labels etc. if needed
                             newSecret.setData(oldSecret.getData());
                         }
-
+                        
+                        this.apiSecretHash = ReconcilerUtils.hashSecretContent(newSecret, "password");
```

Comment:
Will the CC API secret have a key "password"? From what i understand, the secret has the following three keys:
```
cruise-control.apiAdminPassword
cruise-control.apiUserPassword
cruise-control.apiAuthFile
```

Reply:
No, the hashSecretContent will has all values of keys containing the keyword. In this case: cruise-control.apiAdminPassword and apiUserPassword.

Reply:
Whoops, I misunderstood the code, thanks for the explanation, makes sense!